### PR TITLE
<SwDataTable /> component now passes the value for the open-details a…

### PR DESCRIPTION
**Pull request by atloss**
The "open-details" and "item-delete" event on the <SwDataTable /> component are missing the event value.

The "open-details" and "item-delete" event on the component have an event value in order to work with a single entity like navigating to its detail page oder deleting it.

### CODE REVIEW

**Review by jleifeld — CHANGES_REQUESTED** [review]
Thanks for your contribution! Looks good 👍 

Only thing missing for me are the interaction tests which are checking if the correct event values are emitted. If you could add this that would be awesome 🙂

**Comment by atloss** [comment]
Hi @jleifeld,

to be honest, this is my first pull request ever to an open source project - so I am glad for every feedback.

Unfortunately, I am not that familiar with interaction tests. If you could provide some resources for me I am happy to dive deeper into it and adjust the pull request accordingly.

Cheers, Luca

**Comment by jleifeld** [comment]
> Hi @jleifeld,
> 
> to be honest, this is my first pull request ever to an open source project - so I am glad for every feedback.
> 
> Unfortunately, I am not that familiar with interaction tests. If you could provide some resources for me I am happy to dive deeper into it and adjust the pull request accordingly.
> 
> Cheers, Luca

Hey,

no problem. You can start Storybook with these commands:

```
// Install all dependencies
pnpm install

// Run Storybook in development mode
pnpm run --filter meteor-component-library storybook
```

When you are running the Component Library in Development Mode you will see on the left side the interaction tests:

Bildschirmfoto 2024-03-19 um 09 23 19

The interaction tests for the Data Table are in the following file: `/packages/component-library/src/components/table-and-list/sw-data-table/sw-data-table.interactive.stories.ts`

The base story which is used can be find here: `/packages/component-library/src/components/table-and-list/sw-data-table/sw-data-table.stories.ts`

You need to add a small fix in this file. Inside the story template the event listener aren't correct. These three needs to use the values from the args:

```
// Change from
@open-details="openDetails"
@bulk-edit="bulkEdit"
@bulk-delete="bulkDelete"

// To	
@open-details="args.openDetails"
@bulk-edit="args.bulkEdit"
@bulk-delete="args.bulkDelete"
```

Then you can see in the "Actions" tab next to the "Controls" all triggered events. When you click now on "Edit" or some other action which is emitting a event you see the values here.

Now you can add a new test in the `sw-data-table.interactive.stories.ts` file. You can orientate at the `VisualTestEmitReloadEventOnClickingReload` test. This checks if the reload event gets triggered correctly. You can do a similar thing for your events.

One important thing: your test don't need the prefix "VisualTest". This just triggers a additional screenshot test for visual comparision. In your case this is not needed.

More information about the Storybook interaction tests can be found in the Storybook documentation: https://storybook.js.org/docs/writing-tests/interaction-testing

Or if you have other questions feel free to ask 🙂

**Comment by atloss** [comment]
Thank you @jleifeld,

I added another commit with my two interaction tests for the open details and item delete event.

However, I don't think the event data needs to be passed for the bulk edit and bulk delete event, since this can be retrieved from the selection prop.

Cheers,

Luca

File: packages/component-library/src/components/table-and-list/sw-data-table/sw-data-table.interactive.stories.ts
Comment (jleifeld) [inline packages/component-library/src/components/table-and-list/sw-data-table/sw-data-table.interactive.stories.ts:575]: It would be cool if we check if the correct event data is emitted (toHaveBeenCalledWith). Otherwise the test would be still green if I would remove the date from the events

**Review by jleifeld — CHANGES_REQUESTED** [review]

**Comment by jleifeld** [comment]
> Thank you @jleifeld,
> 
> I added another commit with my two interaction tests for the open details and item delete event.
> 
> However, I don't think the event data needs to be passed for the bulk edit and bulk delete event, since this can be retrieved from the selection prop.
> 
> Cheers,
> 
> Luca

You're correct for the bulk edit and delete event. This is not necessary. Tests are looking almost alright. It would be good if the last event check is more specific to make sure that the data is also inside the event.

File: packages/component-library/src/components/table-and-list/sw-data-table/sw-data-table.interactive.stories.ts
Comment (atloss) [reply packages/component-library/src/components/table-and-list/sw-data-table/sw-data-table.interactive.stories.ts]: Gotcha, I oriented at the `VisualTestEnsureSingleSelectionWithoutLoadMore` `itemAdd` event.

**Review by jleifeld — CHANGES_REQUESTED** [review]
Looks good! If you could rebase your PR then I would merge it 👍

File: packages/component-library/src/components/table-and-list/mt-data-table/mt-data-table.interactive.stories.ts
Comment (jleifeld) [inline packages/component-library/src/components/table-and-list/mt-data-table/mt-data-table.interactive.stories.ts:561]: This needs to be renamed to `MtDataTableStory` to fix the issue in the static analysis

**Review by jleifeld — CHANGES_REQUESTED** [review]

File: packages/component-library/src/components/table-and-list/mt-data-table/mt-data-table.interactive.stories.ts
Comment (atloss) [reply packages/component-library/src/components/table-and-list/mt-data-table/mt-data-table.interactive.stories.ts]: @jleifeld You're right. My bad, I am sorry

**Comment by jleifeld** [comment]
@Haberkamp Can you have a look why the Vercel Job does not run and says "Authorization required to deploy"?

**Review by jleifeld — APPROVED** [review]

**Comment by jleifeld** [comment]
@Haberkamp I updated the branch again and the deploy does not run and therefore I can't merge this PR. Can you have a look at this and after authorizing the deployment directly merging it?

**Pull request merged**
